jetty-websocket/websocket-core-tests/src/test/java/org/eclipse/jetty/websocket/core/WebSocketOpenTest.java

```diff
@@ -157,22 +158,22 @@ public void testAsyncOnOpen() throws Exception
         Parser.ParsedFrame frame = receiveFrame(client.getInputStream());
         assertThat(frame.getPayloadAsUTF8(), is("Hello"));
 
-        // But cannot receive
-        client.getOutputStream().write(RawFrameBuilder.buildClose(new CloseStatus(CloseStatus.NORMAL), true));
-        assertFalse(serverHandler.closeLatch.await(1, TimeUnit.SECONDS));
-
-        // Can't demand until open
-        assertThrows(Throwable.class, () -> coreSession.demand(1));
-        client.getOutputStream().write(RawFrameBuilder.buildClose(new CloseStatus(CloseStatus.NORMAL), true));
-        assertFalse(serverHandler.closeLatch.await(1, TimeUnit.SECONDS));
```

Comment:
The code above seems to be nice to retain to enforce the behavior that you can't receive until open, etc.
Any reason it was removed? Seems slightly unrelated to this PR.

Reply:
The change to demand breaks this test. This was testing that `demand()` throws if called before succeeding the onOpen callback, this PR changes it so `demand()` does not throw.

So now if demand is called before the callback, we can receive a frame while still in `CONNECTED` mode and not `OPEN`. So once the frame is received the whole connection is failed.